src/sentry/api/endpoints/group_hashes_split.py

```diff
@@ -211,123 +203,182 @@ def _unsplit_group(group: Group, hash: str, hierarchical_hashes: Optional[Sequen
+    query = (
+        Query("events", Entity("events"))
+        .set_select(
+            [
+                Function("count", [], "event_count"),
+                Function("argMax", [Column("event_id"), Column("timestamp")], "event_id"),
+                Function("max", [Column("timestamp")], "latest_event_timestamp"),
+                # If hierarchical_hashes contains any of the materialized
+                # hashes, find_hash_expr evaluates to the last found index and
+                # arraySlice will give us this hash + the next child hash that
+                # we use in groupby
+                #
+                # If hierarchical_hashes does not contain any of those hashes,
+                # find_hash_expr will return 1 so we start slicing at the beginning.
+                # This can happen when hierarchical_hashes is empty (=>
+                # hash_slice = []), but we also try to recover gracefully from
+                # a hypothetical case where we are missing some hashes in
+                # postgres (unclear how this could be reached).
+                #
+                # We select some intermediate computation values here which we
+                # definetly don't need the results of. It's just temp vars.
+                Function(
+                    # First we find the materialized hash using find_hash_expr,
+                    # and subtract 1 which should be the parent hash if there
+                    # is one. If there isn't, this now can be an out-of-bounds
+                    # access by being 0 (arrays are indexed starting with 1)
+                    "minus",
+                    [find_hash_expr, 1],
+                    "parent_hash_i",
+                ),
+                # We clip the value to be at least 1, this will be where we
+                # start slicing hierarchical_hashes. 0 would be an out of
+                # bounds access.
+                Function("greatest", [Column("parent_hash_i"), 1], "slice_start"),
+                # This will return a slice of length 2 if the materialized hash
+                # has been found at the beginning of the array, but return a
+                # slice of length 3 if not.
+                Function(
+                    "arraySlice",
+                    [
+                        Column("hierarchical_hashes"),
+                        Column("slice_start"),
+                        Function(
+                            "minus",
+                            [
+                                Function(
+                                    "plus",
+                                    [Column("parent_hash_i"), 3],
+                                ),
+                                Column("slice_start"),
+                            ],
+                        ),
+                    ],
+                    "hash_slice",
+                ),
+                Column("primary_hash"),
+            ]
+        )
+        .set_where(_get_group_filters(group))
```

Comment:
I'm not sure if you've tested this yet, but the events entity should error if you don't provide a complete time range (>= and < Condition).

Reply:
yes you're right. I still would like to remove this condition but for now reverted the changes